pkg/kube/controllers/common.go

```diff
@@ -179,23 +184,33 @@ func filteredObjectHandler(handler func(o Object), onlyIncludeSpecChanges bool,
 	}
 }
 
-func extractObject(obj any) Object {
-	o, ok := obj.(Object)
+// Extract pulls a T from obj, handling tombstones.
+// This will return nil if the object cannot be extracted.
+func Extract[T Object](obj any) T {
```

Comment:
We can not get the valid error info from the logs of this function, we need to print type as well

Reply:
👍 will improve the message

Reply:
Added tests and now the log says: `tombstone contained object that is not an object (key:, obj:*v1.Pod)` (note: in real world key will normally be there, this is just from test that doesn't set it)